compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -200,12 +200,13 @@ class ClassfileParser(
         sym.markAbsent()
       }
 
-    // eager load java enum definitions for exhaustivity check of pattern match
+    // eager load enum definitions for exhaustivity check of pattern match
     if (isEnum) {
       instanceScope.toList.map(_.ensureCompleted())
       staticScope.toList.map(_.ensureCompleted())
-      classRoot.setFlag(Flags.JavaEnumTrait)
-      moduleRoot.setFlag(Flags.JavaEnumTrait)
+      val flag = if moduleRoot.is(Flags.JavaDefined) then Flags.JavaEnumTrait else Flags.Enum
+      classRoot.setFlag(flag)
```

Comment:
Why is enum treated specially ? The normal way to set java flags is done above:
```scala
    val sflags       = classTranslation.flags(jflags)
    // ...
    classRoot.setFlag(sflags)
```
It seems that the only missing piece is that [translateFlag](https://github.com/lampepfl/dotty/blob/83ab6c7ee82ff854ee782790d897748d474ccc0d/compiler/src/dotty/tools/dotc/core/classfile/ClassfileConstants.scala#L340) needs to be taught about the JAVA_ACC_ENUM flag (in fact, the corresponding method in scalac already does that: https://github.com/scala/scala/blob/d41fd0cb21edc102a4ab34b03e24c4215d130207/src/reflect/scala/reflect/internal/ClassfileConstants.scala#L356)